### ISSUE TITLE
Wrong error message for overloaded intrinsics

### ISSUE THREAD

**Issue by yskim1501** [microsoft/DirectXShaderCompiler#818]
The following code produces wrong message:

```HLSL
float4 g_f;
int4 g_i;
uint4 g_u;
double4 g_f64;
int64_t4 g_i64;
uint64_t4 g_u64;

float4 main() : SV_Target {
  float4 f1 = 0;
  f1 += asfloat(g_f);
  f1 += asfloat(g_i);
  f1 += asfloat(g_u);
  f1 += asfloat(g_f64);
  f1 += asfloat(g_i64);
  f1 += asfloat(g_u64);   
  return 1;
}
```
```
smoke.hlsl:15:9: error: call to 'asfloat' is ambiguous
  f1 += asfloat(g_f64);
        ^~~~~~~
note: candidate function
note: candidate function
note: candidate function
smoke.hlsl:16:17: warning: conversion from larger type 'const int64_t4' to smaller type 'vector<int, 4>', possible loss of data
  f1 += asfloat(g_i64);
                ^
smoke.hlsl:17:9: error: call to 'asfloat' is ambiguous
  f1 += asfloat(g_u64);
        ^~~~~~~
note: candidate function
note: candidate function
note: candidate function
```
This is because after an intrinsic overload, clang will find a match without asking HLSL external source and implicitly cast types.
